homeassistant/components/powerwall/binary_sensor.py

```diff
@@ -101,7 +102,30 @@ def unique_id(self) -> str:
     @property
     def is_on(self) -> bool:
         """Grid is online."""
-        return self.data.grid_status == GridStatus.CONNECTED
+        return self.data.grid_status in [
+            GridStatus.TRANSITION_TO_GRID,
+            GridStatus.CONNECTED,
+        ]
+
+
+class PowerWallOffGridStatusSensor(PowerWallEntity, BinarySensorEntity):
+    """Representation of an Powerwall off-grid status sensor."""
+
+    _attr_name = "Off-grid Status"
+    _attr_device_class = BinarySensorDeviceClass.POWER
+
+    @property
+    def unique_id(self) -> str:
+        """Device Uniqueid."""
+        return f"{self.base_unique_id}_off_grid_status"
+
+    @property
+    def is_on(self) -> bool:
+        """Grid is offline."""
+        return self.data.grid_status in [
+            GridStatus.TRANSITION_TO_ISLAND,
+            GridStatus.ISLANDED,
+        ]
```

Comment:
```suggestion
```

Since you intend to add a switch I don't think we need a binary sensor as well since the state will always be the same as the switch in https://github.com/home-assistant/core/pull/86357

Reply:
Fair point! I'll remove the Binary sensor from my other PR and close this one then?